trieste/models/gpflow/sampler.py

```diff
@@ -91,11 +113,23 @@ def sample(self, at: TensorType, *, jitter: float = DEFAULTS.JITTER) -> TensorTy
         mean, var = self._model.predict(at[..., None, :, :])  # [..., 1, 1, L], [..., 1, 1, L]
         var = var + jitter
 
-        if not self._initialized:
-            self._eps.assign(
-                tf.random.normal([self._sample_size, tf.shape(mean)[-1]], dtype=tf.float64)
-            )  # [S, L]
+        def sample_eps() -> tf.Tensor:
             self._initialized.assign(True)
+            skip = IndependentReparametrizationSampler.skip
+            IndependentReparametrizationSampler.skip.assign(skip + self._sample_size)
+            normal_samples = idealised_normal_samples(
+                tf.TensorShape(self._sample_size), mean.shape[-1], skip
+            )
+            return normal_samples  # [S, L]
+
+        if self._eps is None:
+            self._eps = tf.Variable(sample_eps())
+
+        tf.cond(
+            self._initialized,
+            lambda: self._eps,
+            lambda: self._eps.assign(sample_eps()),
+        )
 
         return mean + tf.sqrt(var) * tf.cast(self._eps[:, None, :], var.dtype)  # [..., S, 1, L]
```

Comment:
should this be `self._eps[..., None, :]`?

Reply:
I don't think so? `self._eps` is still shape `[S, L]` without any leading dimensions.

Reply:
you're right, sorry